### PR TITLE
clippy: fix build

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by trofi — APPROVED** [review]
Fixes the build on `x86_64-linux` for me. The resulting clippy runs successfully. Thank you!

File: pkgs/development/compilers/rust/clippy.nix [hunk 1]
```diff
@@ -20,6 +20,10 @@ rustPlatform.buildRustPackage {
   #   (/private/tmp/nix-build-clippy-1.36.0.drv-0/rustc-1.36.0-src/src/librustc_llvm)
   doCheck = false;
 
+  postBuild = ''
+    ln -s src/tools/clippy/target target
```
Comment (Mic92) [inline pkgs/development/compilers/rust/clippy.nix:24]:
```suggestion
         export CARGO_TARGET_DIR="$(pwd)/target"
```

**Comment by Mic92** [comment]
@ofborg build clippy

File: pkgs/development/compilers/rust/clippy.nix [hunk 2]
```diff
@@ -20,6 +20,10 @@ rustPlatform.buildRustPackage {
   #   (/private/tmp/nix-build-clippy-1.36.0.drv-0/rustc-1.36.0-src/src/librustc_llvm)
   doCheck = false;
 
+  postBuild = ''
```
Comment (Mic92) [inline pkgs/development/compilers/rust/clippy.nix:23]:
```suggestion
  preBuild = ''
```

**Comment by Mic92** [comment]
@ofborg build clippy

**Review by jdreaver — APPROVED** [review]
This has been a thorn in my side for a few days. Thanks for fixing this!

**Comment by Mic92** [comment]
Once this is merged into staging, we can undo it in staging again.

**Comment by K900** [comment]
Should be all good to revert this now

**Comment by Mic92** [comment]
https://github.com/NixOS/nixpkgs/pull/157295

**Pull request merged**